03week/towersOfHanoi/towersOfHanoi.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const assert = require('assert');
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+let colors = require('colors'); 
+
+let stacks = {
+  a: [4, 3, 2, 1],
+  b: [],
+  c: []
+};
+
+function printStacks() {
+  console.log("a: " + stacks.a);
+  console.log("b: " + stacks.b);
+  console.log("c: " + stacks.c);
+}
+
+// moves last block from startStack to endStack
+
+function movePiece(arr1, arr2) {
+  stacks[`${arr2}`].push(stacks[`${arr1}`].pop()) 
+}
+
+// checks startStack is not empty
+// checks if startStack is less than endStack
+// accounts for undefined < 1 returning from false from line above
+
+function isLegal(arr1, arr2) { 
+  return !stacks[`${arr1}`].length ? false
```

Comment:
nice of ternary here, but I wouldn't recommend nesting ternaries because it starts getting hard to read